presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -283,6 +312,28 @@ else if (column.getBaseHiveColumnIndex() < fileColumns.size()) {
         }
     }
 
+    private static OrcColumn getNestedOrcColumn(OrcColumn baseOrcColumn, Optional<HiveColumnProjectionInfo> partialColumnInfo)
+    {
+        if (!partialColumnInfo.isPresent()) {
+            return baseOrcColumn;
+        }
+
+        OrcColumn current = baseOrcColumn;
+        List<String> dereferenceNames = partialColumnInfo.get().getDereferenceNames();
+        for (int i = 0; i < dereferenceNames.size(); i++) {
+            // Get nested columns by name
+            Map<String, OrcColumn> nestedColumns = current.getNestedColumns().stream()
```

Comment:
No point in building a map to do a single lookup. Iterate over the list manually and find the first element that matches:

```java
OrcColumn current = baseOrcColumn;
for (String field : partialColumnInfo.get().getDereferenceNames()) {
    Optional<OrcColumn> orcColumn = current.getNestedColumns().stream()
            .filter(column -> column.getColumnName().toLowerCase(ENGLISH).equals(field))
            .findFirst();

    if (!orcColumn.isPresent()) {
        return null;
    }

    current = orcColumn.get();
}
```